mailbox/api/src/main/java/org/apache/james/mailbox/model/MessageRange.java

```diff
@@ -46,6 +46,7 @@
     public static final long NOT_A_UID = -1;
 
     public static final long MAX_UID = Long.MAX_VALUE;
+    public static final long UNKNOWN_SIZE = -1;
```

Comment:
After reading your comments, I realized they were no need to handle deletes using MessageRange if I rework ListeningSearchIndex class a little. I deleted this changeset and will propose a new one, of higher quality.